core/include/detray/propagator/propagator.hpp

```diff
@@ -139,31 +134,35 @@ struct propagator {
     /// @return propagation success.
     template <typename state_t, typename actor_states_t = actor_chain<>::state>
     DETRAY_HOST_DEVICE bool propagate(state_t &propagation,
-                                      actor_states_t &&actor_states = {}) {
+                                      actor_states_t &&actor_states = {},
+                                      const config &cfg = {}) {
```

Comment:
I understand the intention but wouldn't it be better to have config as a member in propagator rather than taking it as an argument for `propagate` function

Reply:
Done. I guess, this has the disadvantage though that if you add the ```detray::propagation::config``` to another algorithm config in traccc, it has to be copied one more time into the propagator now, instead of just being referenced from the propagation call